internal/encoded.go

```diff
@@ -71,22 +71,32 @@ type (
 		ToData(value ...interface{}) (*commonpb.Payloads, error)
 		// FromData implements conversion of an array of values of different types.
 		// Useful for deserializing arguments of function invocations.
-		FromData(input *commonpb.Payloads, valuePtr ...interface{}) error
+		FromData(input *commonpb.Payloads, valuePtrs ...interface{}) error
 	}
 
-	// defaultDataConverter uses JSON.
-	defaultDataConverter struct{}
+	// PayloadConverter converts single value to/from payload.
+	PayloadConverter interface {
+		// ToData single value to payload.
+		ToData(value interface{}) (*commonpb.Payload, error)
+		// FromData single value from payload.
+		FromData(input *commonpb.Payload, valuePtr interface{}) error
+	}
+
+	defaultPayloadConverter struct{}
```

Comment:
For now this is the only one implementation. And name is not good. I will refactor it and will make it plugable later.